pennylane/math/utils.py

```diff
@@ -279,6 +279,39 @@ def _get_interface_of_single_tensor(tensor):
     return res
 
 
+def get_deep_interface(value):
+    """
+    Given a deep data structure with interface-specific scalars at the bottom, return their
+    interface name.
+
+    Args:
+        value (list, tuple): A deep list-of-lists, tuple-of-tuples, or combination with
+            interface-specific data hidden within it
+
+    Returns:
+        str: The name of the interface deep within the value
+
+    **Example**
+
+    >>> x = [[jax.numpy.array(1), jax.numpy.array(2)], [jax.numpy.array(3), jax.numpy.array(4)]]
+    >>> get_deep_interface(x)
+    'jax'
+
+    This can be especially useful when converting to the appropriate interface:
+
+    >>> qml.math.asarray(x, like=qml.math.get_deep_interface(x))
+    Array([[1, 2],
+       [3, 4]], dtype=int64)
+
+    """
+    itr = value
+    while isinstance(itr, (list, tuple)):
+        if len(itr) == 0:
+            return "builtins"
+        itr = itr[0]
+    return ar.infer_backend(itr)

```

Comment:
What about something like `[2, jnp.array(2)]`?

Reply:
my goal was not to make a universal solution here, so this won't work. I mainly wrote this because we often have lists of data. if you have something like this, I can't help you:
```python
x = jnp.arange(4)
data = [x.shape, x]
qml.math.ndim(data)
```
^ I can make something more generic, but I don't really want to unless necessary.